src/GitHub.App/Services/ModelService.cs

```diff
@@ -480,8 +480,20 @@ public PullRequestCacheItem(PullRequest pr, IReadOnlyList<PullRequestFile> files
             {
                 Title = pr.Title;
                 Number = pr.Number;
-                Base = new GitReferenceCacheItem { Label = pr.Base.Label, Ref = pr.Base.Ref, RepositoryCloneUrl = pr.Base.Repository.CloneUrl };
-                Head = pr.Head != null ? new GitReferenceCacheItem { Label = pr.Head.Label, Ref = pr.Head.Ref, RepositoryCloneUrl = pr.Head.Repository.CloneUrl } : null;
+                Base = new GitReferenceCacheItem
+                {
+                    Label = pr.Base.Label,
+                    Ref = pr.Base.Ref,
+                    Sha = pr.Base.Sha,
+                    RepositoryCloneUrl = pr.Base.Repository.CloneUrl,
+                };
+                Head = pr.Head != null ? new GitReferenceCacheItem
+                {
+                    Label = pr.Head.Label,
+                    Ref = pr.Head.Ref,
+                    Sha = pr.Head.Sha,
+                    RepositoryCloneUrl = pr.Head.Repository.CloneUrl
```

Comment:
If the user has deleted the repository after submitting the PR, `pr.Head.Repository` will be null even if `pr.Head` isn't (the PR has the info, just not the data for the actual repo), so this requires handling for null.